### PR TITLE
Feature/risk rule plugin ( WIP )

### PR DESCRIPTION
**Pull request by Blokh**

### CODE REVIEW

File: services/workflows-service/src/business/business.module.ts [hunk 1]
```diff
@@ -39,6 +40,7 @@ import { BusinessReportModule } from '@/business-report/business-report.module';
     ProjectModule,
     CustomerModule,
     forwardRef(() => BusinessReportModule),
+    RuleEngineModule,
```
Comment (liorzam) [inline services/workflows-service/src/business/business.module.ts:43]: why?

File: services/workflows-service/src/env.ts [hunk 2]
```diff
@@ -88,7 +88,7 @@ if (!process.env['ENVIRONMENT_NAME'] || process.env['ENVIRONMENT_NAME'] === 'loc
     }
   }
 
-  console.log('Environment variables loaded', severEnvVars);
+  // console.log('Environment variables loaded', severEnvVars);
```
Comment (liorzam) [inline services/workflows-service/src/env.ts]: its on purpose for local envs, revet

File: services/workflows-service/src/rule-engine/risk-rule.service.ts [hunk 3]
```diff
@@ -1,23 +1,29 @@
...
+  'LT',
+  'GTE',
+  'LTE',
+  'LAST_YEAR',
+] as const satisfies readonly TOperation[];
 
 const RuleSchema = z.object({
   key: z.string(),
-  operation: z.enum(['equals', 'not_equals', 'contains', 'not_contains']),
+  operation: z.enum(operations),
```
Comment (liorzam) [inline services/workflows-service/src/rule-engine/risk-rule.service.ts]: GJ

File: services/workflows-service/src/rule-engine/risk-rule.service.ts [hunk 4]
```diff
@@ -1,23 +1,29 @@
...
 type Rule = z.infer<typeof RuleSchema>;
 
-type RuleSet = {
-  operator?: 'and' | 'or';
-  rules: Array<RuleSet | Rule>;
-};
-
 const RuleSetSchema: z.ZodType<RuleSet> = z.object({
-  operator: z.enum(['and', 'or']).optional(),
+  operator: z.enum(['and', 'or']),
```
Comment (liorzam) [inline services/workflows-service/src/rule-engine/risk-rule.service.ts]: also should be from enum

File: services/workflows-service/src/rule-engine/rule-engine.service.ts [hunk 5]
```diff
@@ -1,15 +1,11 @@
 import { Injectable } from '@nestjs/common';
-import { RuleStoreService, RuleStoreServiceFindAllOptions } from './rule-store.service';
 import { operationHelpers } from './core/operators/constants';
 import { RuleEngine } from './core/rule-engine';
+import { RuleSet } from '@/rule-engine/core/types';
 
 @Injectable()
 export class RuleEngineService {
-  constructor(private readonly ruleStoreService: RuleStoreService) {}
```
Comment (liorzam) [inline services/workflows-service/src/rule-engine/rule-engine.service.ts:8]: can you do some printing using our logger on engine behalf ?

File: services/workflows-service/src/workflow/workflow.service.ts [hunk 6]
```diff
@@ -1923,6 +1928,32 @@ export class WorkflowService {
...
+                ...rule,
+              };
+            } catch (ex) {
+              return {
+                ...rule,
+                result: {
+                  status: 'FAILED',
+                  message: isErrorWithMessage(ex) ? ex.message : undefined,
+                  error: ex,
+                },
```
Comment (liorzam) [inline services/workflows-service/src/workflow/workflow.service.ts:1949]: you should have everything you need on the output object ...
lets set this trycatch in the rule engine service

File: services/workflows-service/src/rule-engine/risk-rule.service.ts [hunk 7]
```diff
@@ -58,14 +64,19 @@ const NotionRiskRuleRecordSchema = z
...
 export class RiskRuleService {
   constructor(
     private readonly notionService: NotionService,
     private readonly logger: AppLoggerService,
   ) {}
 
-  public async findAll({ databaseId, source }: { databaseId: string; source: 'notion' }) {
+  public async findAll({ databaseId, source }: TFindAllRulesOptions) {
     if (source === 'notion') {
       const records = await this.notionService.getAllDatabaseRecordsValues({
```
Comment (liorzam) [inline services/workflows-service/src/rule-engine/risk-rule.service.ts:83]: As discussed once the data is present we wont fetch it from remote again

File: services/workflows-service/src/rule-engine/risk-rule.service.ts [hunk 8]
```diff
@@ -58,14 +64,19 @@ const NotionRiskRuleRecordSchema = z
...
+
 @Injectable()
 export class RiskRuleService {
   constructor(
     private readonly notionService: NotionService,
     private readonly logger: AppLoggerService,
   ) {}
 
-  public async findAll({ databaseId, source }: { databaseId: string; source: 'notion' }) {
+  public async findAll({ databaseId, source }: TFindAllRulesOptions) {
```
Comment (liorzam) [inline services/workflows-service/src/rule-engine/risk-rule.service.ts:81]: Please inject sentry service and trigger alert when couldnt parse rule

Comment (Blokh) [reply services/workflows-service/src/rule-engine/risk-rule.service.ts:81]: isn't appLogger has a connection to sentry?

File: services/workflows-service/src/workflow/workflow.service.ts (hunk 6 above)
Comment (Blokh) [reply services/workflows-service/src/workflow/workflow.service.ts:1949]: it's for specific rule failing

File: services/workflows-service/src/business/business.module.ts (hunk 1 above)
Comment (Blokh) [reply services/workflows-service/src/business/business.module.ts:43]: remove it and see what happens

**Pull request merged**